teos/src/main.rs

```diff
@@ -328,28 +324,28 @@ async fn main() {
         .client_ca_root(Certificate::from_pem(ca_cert));
 
     // Start tasks
-    let private_api_task = task::spawn(async move {
+    let private_api_task = task::spawn(enclose!((internal_api) async move {
```

Comment:
For some reason my opinion has changed about this :joy:
I think we better just do
```rust
let internal_api_private_task = internal_api.clone();
```
Instead of using the macro for this.

macros may look hard to some new comers and this piece might not make much sense.
tried to think of a more descriptive name that would make sense, but it's gonna be a very long name xD